stl/inc/cmath

```diff
@@ -953,261 +952,373 @@ _NODISCARD _CRT_SATELLITE_2 float __stdcall __std_smf_hypot3f(float, float, floa
     return __std_smf_sph_besself(_Arg1, _Arg2);
 }
 
-_NODISCARD inline long double sph_bessell(const unsigned int _Arg1, const long double _Arg2) {
+_NODISCARD inline long double sph_bessell(const unsigned int _Arg1, const long double _Arg2) noexcept
+/* strengthened */ {
     return __std_smf_sph_bessel(_Arg1, static_cast<double>(_Arg2));
 }
 
-_NODISCARD inline double sph_legendre(const unsigned int _Arg1, const unsigned int _Arg2, const double _Theta) {
+_NODISCARD inline double sph_legendre(const unsigned int _Arg1, const unsigned int _Arg2, const double _Theta) noexcept
+/* strengthened */ {
     return __std_smf_sph_legendre(_Arg1, _Arg2, _Theta);
 }
 
-_NODISCARD inline float sph_legendref(const unsigned int _Arg1, const unsigned int _Arg2, const float _Theta) {
+_NODISCARD inline float sph_legendref(const unsigned int _Arg1, const unsigned int _Arg2, const float _Theta) noexcept
+/* strengthened */ {
     return __std_smf_sph_legendref(_Arg1, _Arg2, _Theta);
 }
 
 _NODISCARD inline long double sph_legendrel(
-    const unsigned int _Arg1, const unsigned int _Arg2, const long double _Theta) {
+    const unsigned int _Arg1, const unsigned int _Arg2, const long double _Theta) noexcept /* strengthened */ {
     return __std_smf_sph_legendre(_Arg1, _Arg2, static_cast<double>(_Theta));
 }
 
-_NODISCARD inline double sph_neumann(const unsigned int _Arg1, const double _Arg2) {
+_NODISCARD inline double sph_neumann(const unsigned int _Arg1, const double _Arg2) noexcept /* strengthened */ {
     return __std_smf_sph_neumann(_Arg1, _Arg2);
 }
 
-_NODISCARD inline float sph_neumannf(const unsigned int _Arg1, const float _Arg2) {
+_NODISCARD inline float sph_neumannf(const unsigned int _Arg1, const float _Arg2) noexcept /* strengthened */ {
     return __std_smf_sph_neumannf(_Arg1, _Arg2);
 }
 
-_NODISCARD inline long double sph_neumannl(const unsigned int _Arg1, const long double _Arg2) {
+_NODISCARD inline long double sph_neumannl(const unsigned int _Arg1, const long double _Arg2) noexcept
+/* strengthened */ {
     return __std_smf_sph_neumann(_Arg1, static_cast<double>(_Arg2));
 }
+_STD_END
 
+#define _GENERIC_MATH_SPECIAL1(NAME)                                                  \
+    _STD_BEGIN                                                                        \
+    template <class _Ty, enable_if_t<is_arithmetic_v<_Ty>, int> = 0>                  \
+    _NODISCARD auto NAME(const _Ty _Arg) noexcept /* strengthened */ {                \
+        using _Common = conditional_t<is_integral_v<_Ty>, double, _Ty>;               \
+        if constexpr (is_same_v<_Common, float>)                                      \
+            return __std_smf_##NAME##f(_Arg);                                         \
+        else                                                                          \
+            return static_cast<_Common>(__std_smf_##NAME(static_cast<double>(_Arg))); \

```

Comment:
We conventionally fully-brace `if` and `else`, even when the _compound-statement_ is a single statement. (Also on 1245, 1258, 1271, and 1284.